src/core/ipns/validator.js

```diff
@@ -33,10 +35,12 @@ const verify = (publicKey, entry, callback) => {
     }
 
     // Validate EOL
-    const validityDate = Date.parse(validity.toString())
+    if (validityType === IpnsEntry.validityType.EOL) {
```

Comment:
We should fail by default if we can't validate the record.

Reply:
already done in [js-ipns](https://github.com/ipfs/js-ipns), more precisely [here](https://github.com/ipfs/js-ipns/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638R74)